src/mesh/mesh_adaptation.cpp

```diff
@@ -8,8 +9,10 @@ MeshAdaptation<dim,real,MeshType>::MeshAdaptation(std::shared_ptr< DGBase<dim, r
     , critical_residual(dg->all_parameters->mesh_adaptation_param.critical_residual_val)
     , total_refinement_cycles(dg->all_parameters->mesh_adaptation_param.total_refinement_steps)
     , current_refinement_cycle(0)
-    , refinement_fraction(dg->all_parameters->mesh_adaptation_param.refinement_fraction)
-    , coarsening_fraction(dg->all_parameters->mesh_adaptation_param.coarsening_fraction)
+    , h_refine_fraction(dg->all_parameters->mesh_adaptation_param.h_refine_fraction)
+    , h_coarsen_fraction(dg->all_parameters->mesh_adaptation_param.h_coarsen_fraction)
+    , p_refine_fraction(dg->all_parameters->mesh_adaptation_param.p_refine_fraction)
+    , p_coarsen_fraction(dg->all_parameters->mesh_adaptation_param.p_coarsen_fraction)
```

Comment:
At this point, instead of passing all parameters through DG, I think it's worth passing the MeshAdaptationParameter.

Reply:
Maybe even just store the MeshAdaptation parameter.